src/lib/datatypes/sol-buffer.c

```diff
@@ -344,6 +347,20 @@ sol_buffer_steal(struct sol_buffer *buf, size_t *size)
     return r;
 }
 
+SOL_API void *
+sol_buffer_steal_or_copy(struct sol_buffer *buf, size_t *size)
```

Comment:
you can add this as a `static inline` in the header, this way we keep the function without generating an extra symbol